homeassistant/components/lyric/climate.py

```diff
@@ -356,56 +366,69 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set hvac mode."""
         _LOGGER.debug("HVAC mode: %s", hvac_mode)
         try:
-            if LYRIC_HVAC_MODES[hvac_mode] == LYRIC_HVAC_MODE_HEAT_COOL:
-                # If the system is off, turn it to Heat first then to Auto,
-                # otherwise it turns to.
-                # Auto briefly and then reverts to Off (perhaps related to
-                # heatCoolMode). This is the behavior that happens with the
-                # native app as well, so likely a bug in the api itself
-                if HVAC_MODES[self.device.changeableValues.mode] == HVACMode.OFF:
-                    _LOGGER.debug(
-                        "HVAC mode passed to lyric: %s",
-                        HVAC_MODES[LYRIC_HVAC_MODE_COOL],
-                    )
-                    await self._update_thermostat(
-                        self.location,
-                        self.device,
-                        mode=HVAC_MODES[LYRIC_HVAC_MODE_HEAT],
-                        autoChangeoverActive=False,
-                    )
-                    # Sleep 3 seconds before proceeding
-                    await asyncio.sleep(3)
-                    _LOGGER.debug(
-                        "HVAC mode passed to lyric: %s",
-                        HVAC_MODES[LYRIC_HVAC_MODE_HEAT],
-                    )
-                    await self._update_thermostat(
-                        self.location,
-                        self.device,
-                        mode=HVAC_MODES[LYRIC_HVAC_MODE_HEAT],
-                        autoChangeoverActive=True,
-                    )
-                else:
-                    _LOGGER.debug(
-                        "HVAC mode passed to lyric: %s",
-                        HVAC_MODES[self.device.changeableValues.mode],
-                    )
-                    await self._update_thermostat(
-                        self.location, self.device, autoChangeoverActive=True
-                    )
-            else:
+            match self._attr_thermostat_type:
+                case LyricThermostatType.TCC:
+                    await self._async_set_hvac_mode_tcc(hvac_mode)
+                case LyricThermostatType.LCC:
+                    await self._async_set_hvac_mode_lcc(hvac_mode)
+        except LYRIC_EXCEPTIONS as exception:
+            _LOGGER.error(exception)
+        await self.coordinator.async_refresh()
```

Comment:
Why do we refresh the coordinator after an error?

Reply:
Good question. This does seem unnecessary. Similar to my other comment though, I'm just chameleon-coding here: trying to make my changes blend in with the existing code. If I were to make a separate PR like I suggested in my other comment to raise exceptions instead of logging errors, then this wouldn't be a problem anymore because we'd exit before getting to the refresh call.

Reply:
Check with codeowners if a refresh is needed. Otherwise we can just remove it here.

Reply:
Happy to do so, but like the other comment, this behavior existed already. I'd prefer to make that change in a separate PR if possible instead of blocking this change.

Reply:
Sure.